src/CoreFoundation/CFProxySupport.cs

```diff
@@ -579,62 +581,241 @@ public static CFProxySettings GetSystemProxySettings ()
 			dict.DangerousRelease ();
 			return new CFProxySettings (dict);
 		}
-		
-#if notyet
-		// FIXME: These require bindings for CFRunLoopSource and CFStreamClientContext
-		
-		public delegate void CFProxyAutoConfigurationResultCallback (IntPtr client, NSArray proxyList, NSError error);
-		
+
+		delegate void CFProxyAutoConfigurationResultCallbackInternal (IntPtr client, IntPtr proxyList, IntPtr error);
+		public delegate void CFProxyAutoConfigurationResultCallback (NSObject client, CFProxy[] proxyList, NSError error);
+		// helper delegate to reuse code
+		delegate IntPtr CreatePACCFRunLoopSource (CFProxyAutoConfigurationResultCallbackInternal cb, ref CFStreamClientContext context);
+
+		static CFProxy[] ParseProxies (IntPtr proxyList) {
+			CFProxy[] proxies = null;
+			if (proxyList != IntPtr.Zero) {
+				using (var array = new CFArray (proxyList, false)){
+					proxies = new CFProxy [array.Count];
+					for (int i = 0; i < proxies.Length; i++) {
+						var dict = new NSDictionary (array.GetValue (i));
```

Comment:
you should use `CFDictionary` inside CoreFoundation'
even more if you think this code could move (like in mono) in the future

Reply:
There is no constructor for CFProxy that takes a CFDictionary :/

Reply:
it's an internal API so it could be changed - but since it's already existing (and not new to this PR) let's not introduce more changes than are required :)